### PR TITLE
Fix panic when storage value is nil

### LINKED ISSUE #5425: Vault panics when attempting to enable KV versioning
**Describe the bug**
I had an old Vault 0.8.3 server setup which I recently upgraded to 0.11.1. No problems there.

Now we want to implement KV versioning at our `secret/` path which is currently like so:

```
vault secrets list -format=json
[...]
  "secret/": {
    "type": "kv",
    "description": "key/value secret storage",
    "accessor": "kv_bb0c0f3f",
    "config": {
      "default_lease_ttl": 0,
      "max_lease_ttl": 0,
      "force_no_cache": false
    },
    "options": null,
    "local": false,
    "seal_wrap": false
  },
[...]
```
so I ran the command `vault kv enable-versioning secret/`. I received back `Success! Tuned the secrets engine at: secret/`; however, turns out Vault had actually panic'ed:

```
2018-09-27T12:57:29.402-0400 [INFO ] core: mount tuning of options: path=secret/ options=map[version:2]
2018-09-27T12:57:29.405-0400 [INFO ] secrets.kv.kv_bb0c0f3f: collecting keys to upgrade
2018-09-27T12:57:29.446-0400 [INFO ] secrets.kv.kv_bb0c0f3f: done collecting keys: num_keys=306
panic: runtime error: slice bounds out of range

goroutine 704 [running]:
github.com/hashicorp/vault/vault.(*AESGCMBarrier).Get(0xc0005dd500, 0x5613a11eefe0, 0xc00003e0a8, 0xc0008774c0, 0x34, 0x0, 0x0, 0x0)
        /build/vault/src/src/github.com/hashicorp/vault/vault/barrier_aes_gcm.go:712 +0x627
github.com/hashicorp/vault/vault.(*BarrierView).Get(0xc00050c820, 0x5613a11eefe0, 0xc00003e0a8, 0xc000876cf3, 0x7, 0x0, 0x0, 0x0)
        /build/vault/src/src/github.com/hashicorp/vault/vault/barrier_view.go:72 +0xdd
github.com/hashicorp/vault/vendor/github.com/hashicorp/vault-plugin-secrets-kv.(*versionedKVBackend).Upgrade.func2(0xc000876cf3, 0x7, 0x0, 0x0)
        /build/vault/src/src/github.com/hashicorp/vault/vendor/github.com/hashicorp/vault-plugin-secrets-kv/upgrade.go:101 +0xba
github.com/hashicorp/vault/vendor/github.com/hashicorp/vault-plugin-secrets-kv.(*versionedKVBackend).Upgrade.func3(0x5613a11efd20, 0xc00050c820, 0xc000abff80, 0xc000993b90, 0xc000becb70, 0xe, 0xe, 0xc0005df5c0, 0xc000abff90)
        /build/vault/src/src/github.com/hashicorp/vault/vendor/github.com/hashicorp/vault-plugin-secrets-kv/upgrade.go:188 +0x71b
created by github.com/hashicorp/vault/vendor/github.com/hashicorp/vault-plugin-secrets-kv.(*versionedKVBackend).Upgrade
        /build/vault/src/src/github.com/hashicorp/vault/vendor/github.com/hashicorp/vault-plugin-secrets-kv/upgrade.go:155 +0x278
```
I restarted Vault, but as soon as I unsealed it it panic'ed again:
```
2018-09-27T12:59:07.973-0400 [INFO ] core: vault is unsealed
2018-09-27T12:59:07.973-0400 [INFO ] core: entering standby mode
2018-09-27T12:59:07.976-0400 [INFO ] core: acquired lock, enabling active operation
2018-09-27T12:59:08.007-0400 [INFO ] core: post-unseal setup starting
2018-09-27T12:59:08.008-0400 [INFO ] core: loaded wrapping token key
2018-09-27T12:59:08.008-0400 [INFO ] core: successfully setup plugin catalog: plugin-directory=
2018-09-27T12:59:08.009-0400 [INFO ] core: successfully mounted backend: type=kv path=secret/
2018-09-27T12:59:08.009-0400 [INFO ] core: successfully mounted backend: type=system path=sys/
2018-09-27T12:59:08.009-0400 [INFO ] core: successfully mounted backend: type=ssh path=kaloom-signer/
2018-09-27T12:59:08.010-0400 [INFO ] core: successfully mounted backend: type=identity path=identity/
2018-09-27T12:59:08.010-0400 [INFO ] core: successfully mounted backend: type=cubbyhole path=cubbyhole/
2018-09-27T12:59:08.012-0400 [INFO ] core: restoring leases
2018-09-27T12:59:08.012-0400 [INFO ] rollback: starting rollback manager
2018-09-27T12:59:08.035-0400 [INFO ] identity: entities restored
2018-09-27T12:59:08.037-0400 [INFO ] identity: groups restored
2018-09-27T12:59:08.038-0400 [INFO ] core: post-unseal setup complete
2018-09-27T12:59:08.040-0400 [INFO ] core: core/startClusterListener: starting listener: listener_address=127.0.0.1:8201
2018-09-27T12:59:08.040-0400 [INFO ] core: core/startClusterListener: serving cluster requests: cluster_listen_address=127.0.0.1:8201
2018-09-27T12:59:08.049-0400 [INFO ] secrets.kv.kv_bb0c0f3f: collecting keys to upgrade
2018-09-27T12:59:08.145-0400 [INFO ] expiration: lease restore complete
2018-09-27T12:59:08.180-0400 [INFO ] secrets.kv.kv_bb0c0f3f: done collecting keys: num_keys=342
panic: runtime error: slice bounds out of range

goroutine 68 [running]:
github.com/hashicorp/vault/vault.(*AESGCMBarrier).Get(0xc0005df560, 0x5640c908afe0, 0xc00003e0a8, 0xc00089c740, 0x34, 0x0, 0x0, 0x0)
        /build/vault/src/src/github.com/hashicorp/vault/vault/barrier_aes_gcm.go:712 +0x627
github.com/hashicorp/vault/vault.(*BarrierView).Get(0xc000632fa0, 0x5640c908afe0, 0xc00003e0a8, 0xc0000449f3, 0x7, 0x0, 0x5640ca94e1a0, 0x3)
        /build/vault/src/src/github.com/hashicorp/vault/vault/barrier_view.go:72 +0xdd
github.com/hashicorp/vault/vendor/github.com/hashicorp/vault-plugin-secrets-kv.(*versionedKVBackend).Upgrade.func2(0xc0000449f3, 0x7, 0x0, 0x0)
        /build/vault/src/src/github.com/hashicorp/vault/vendor/github.com/hashicorp/vault-plugin-secrets-kv/upgrade.go:101 +0xba
github.com/hashicorp/vault/vendor/github.com/hashicorp/vault-plugin-secrets-kv.(*versionedKVBackend).Upgrade.func3(0x5640c908bd20, 0xc000632fa0, 0xc000732bd0, 0xc0006beaf0, 0xc00067f370, 0xd, 0xd, 0xc000735ec0, 0xc000732be0)
        /build/vault/src/src/github.com/hashicorp/vault/vendor/github.com/hashicorp/vault-plugin-secrets-kv/upgrade.go:188 +0x71b
created by github.com/hashicorp/vault/vendor/github.com/hashicorp/vault-plugin-secrets-kv.(*versionedKVBackend).Upgrade
        /build/vault/src/src/github.com/hashicorp/vault/vendor/github.com/hashicorp/vault-plugin-secrets-kv/upgrade.go:155 +0x278
```
So as far as I can tell my Vault is hosed.  Thankfully, I actually ran this command on a test Vault instance based off of a Consul backup, so the production instance is still fine.

**Expected behavior**
No panic. An error if this is not a supported operation on old secrets instances.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 0.11.1
* Vault CLI Version (retrieve with `vault version`): 0.11.1
* Server Operating System/Architecture: Linux 64-bit

### PR DESCRIPTION
**Pull request by vishalnayak**
Fixes #5425

### CODE REVIEW

File: vault/barrier_aes_gcm.go [hunk 1]
```diff
@@ -708,6 +708,11 @@ func (b *AESGCMBarrier) Get(ctx context.Context, key string) (*Entry, error) {
 		return nil, nil
 	}
 
+	if pe.Value == nil {
```
Comment (kalafut) [inline vault/barrier_aes_gcm.go]: Or should the condition be `len(pe.Value) < 4`?

Comment (vishalnayak) [reply vault/barrier_aes_gcm.go]: I think we should check for both. Added another check.

File: vault/barrier_aes_gcm.go [hunk 2]
```diff
@@ -74,9 +74,9 @@ type AESGCMBarrier struct {
 // the provided physical backend for storage.
 func NewAESGCMBarrier(physical physical.Backend) (*AESGCMBarrier, error) {
 	b := &AESGCMBarrier{
-		backend:                  physical,
```
Comment (jefferai) [inline vault/barrier_aes_gcm.go]: Please check what `gofmt` binary you're using.

Comment (jefferai) [reply vault/barrier_aes_gcm.go]: Or your editor config

File: vault/barrier_aes_gcm.go (hunk 1 above)
Comment (jefferai) [reply vault/barrier_aes_gcm.go]: You don't need both. `len(pe.Value) < 4` will also match when nil.

Comment (briankassouf) [reply vault/barrier_aes_gcm.go]: Yeah i think all we need is `len(pe.Value) < 4`

File: vault/barrier_aes_gcm.go (hunk 2 above)
Comment (vishalnayak) [reply vault/barrier_aes_gcm.go]: Editor was using `goimports` as a format command. Replaced it with `gofmt` now.

File: vault/barrier_aes_gcm.go (hunk 1 above)
Comment (vishalnayak) [reply vault/barrier_aes_gcm.go]: Yeah, right. Fixed.

**Pull request merged**